astroid/interpreter/_import/spec.py

```diff
@@ -162,7 +162,7 @@ def find_module(
 
         for entry in submodule_path:
             package_directory = os.path.join(entry, modname)
-            for suffix in (".py", ".pyi", importlib.machinery.BYTECODE_SUFFIXES[0]):
+            for suffix in (".pyi", ".py", importlib.machinery.BYTECODE_SUFFIXES[0]):
```

Comment:
```suggestion
            # We're looping on pyi first because if a pyi exists there's probably a reason
            # (i.e. the code is hard or impossible to parse), so we take pyi into account
            for suffix in (".pyi", ".py", importlib.machinery.BYTECODE_SUFFIXES[0]):
```